src/hotspot/share/jvmci/jvmciCompilerToVM.cpp

```diff
@@ -1405,6 +1405,13 @@ C2V_END
 
 
 C2V_VMENTRY(void, invalidateHotSpotNmethod, (JNIEnv* env, jobject, jobject hs_nmethod, jboolean deoptimize, jint invalidation_reason))
+#ifdef ASSERT
```

Comment:
We prefer runtime checks and throwing Java exceptions that assertions in this JVMCI code:
```
  int first = static_cast<int>(nmethod::InvalidationReason::UNKNOWN);
  int last = static_cast<int>(nmethod::InvalidationReason::LAST_REASON);
  if (invalidation_reason < first || invalidation_reason >= last) {
    JVMCI_THROW_MSG(IllegalArgumentException, err_msg("Invalid invalidation_reason: %d", invalidation_reason ));
  }
```

Reply:
Got it. I'll patch the code.